functions/go/enforce-gatekeeper/validate.go

```diff
@@ -175,7 +174,75 @@ func parseResults(results []*opatypes.Result) (*framework.Result, error) {
 			}
 		}
 
-		out.Items = append(out.Items, item)
+		items = append(items, item)
+	}
+	sortResultItems(items)
+
+	return &framework.Result{
+		Items: items,
+	}, nil
+}
+
+// TODO(mengqiy): upstream this to the SDK
+func sortResultItems(items []framework.Item) {
+	sort.SliceStable(items, func(i, j int) bool {
+		if severityLess(items, i, j) != 0 {
+			if severityLess(items, i, j) < 0 {
+				return true
+			} else {
+				return false
+			}
+		}
+
+		if fileLess(items, i, j) != 0 {
+			if fileLess(items, i, j) < 0 {
+				return true
+			} else {
+				return false
+			}
+		}
+
+		return resultItemToString(items[i]) < resultItemToString(items[j])
+	})
+}
+
+func severityLess(items []framework.Item, i, j int) int {
+	var severityLevelI, severityLevelJ int
+	switch items[i].Severity {
+	case framework.Error:
+		severityLevelI = 0
+	case framework.Warning:
+		severityLevelI = 1
+	case framework.Info:
+		severityLevelI = 2
+	default:
+		severityLevelI = 3
+	}
```

Comment:
nit: We can have a map for `severity => number` or little function to avoid duplication in this fn.